ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -565,6 +566,87 @@ def test_ipa_healthcheck_ds_ruv_check(self):
             ruvs.remove(check["kw"]["ruv"])
         assert not ruvs
 
+    def test_ipa_healthcheck_tomcatfilecheck(self):
+        """
+        This testcase changes the permissions of the tomcat configuration file
+        on an IPA Master and then checks if healthcheck tools reports the ERROR
+        and when the permissions are reverted back to normal the TomcatFileCheck
+        in ipahealthcheck.ipa.files displays the result as SUCCESS
+        """
+        error_msg1 = (
+            "Permissions of "
+            + paths.PKI_TOMCAT_PASSWORD_CONF
+            + " are too restrictive: 0600 and should be 0660"
+        )
+        error_msg2 = (
+            "Permissions of "
+            + paths.PKI_TOMCAT_SERVER_XML
+            + " are too restrictive: 0600 and should be 0660"
+        )
+        error_msg3 = (
+            "Permissions of "
+            + paths.CA_CS_CFG_PATH
+            + " are too restrictive: 0600 and should be 0660"
+        )
+        self.master.run_command(["chmod", "600", paths.CA_CS_CFG_PATH])
+        self.master.run_command(
+            ["chmod", "600", paths.PKI_TOMCAT_PASSWORD_CONF]
+        )
+        self.master.run_command(["chmod", "600", paths.PKI_TOMCAT_SERVER_XML])
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.files", "TomcatFileCheck"
+        )
+        assert returncode == 1
+        for check in data:
+            if check["kw"]["key"] == "_etc_pki_pki-tomcat_password.conf_mode":
+                assert check["kw"]["path"] == paths.PKI_TOMCAT_PASSWORD_CONF
+                assert check["kw"]["expected"] == "0660"
+                assert check["kw"]["got"] == "0600"
+                assert check["result"] == "ERROR"
+                assert check["kw"]["msg"] == error_msg1
+            elif check["kw"]["key"] == "_etc_pki_pki-tomcat_server.xml_mode":
+                assert check["result"] == "ERROR"
+                assert check["kw"]["path"] == paths.PKI_TOMCAT_SERVER_XML
+                assert check["kw"]["expected"] == "0660"
+                assert check["kw"]["got"] == "0600"
+                assert check["kw"]["msg"] == error_msg2
+            elif (
+                check["kw"]["key"]
+                == "_var_lib_pki_pki-tomcat_conf_ca_CS.cfg_mode"
+            ):
+                assert check["result"] == "ERROR"
+                assert check["kw"]["path"] == paths.CA_CS_CFG_PATH
+                assert check["kw"]["expected"] == "0660"
+                assert check["kw"]["got"] == "0600"
+                assert check["kw"]["msg"] == error_msg3
+        else:
```

Comment:
As the for loop doesn't have any break clause, I find the "else" rather confusing.